### PR TITLE
Uberon curation fixes

### LINKED ISSUE #58: KAZU not matching common GO Biological Process terms
For example, "DNA damage response" or "DNA repair" are not matching despite being very common terms.

### PR DESCRIPTION
**Pull request by RichJackson**
added a new AutoCurationAction and adjusted a few curations. closes #58

### CODE REVIEW

File: kazu/ontology_preprocessing/autocuration.py [hunk 1]
```diff
@@ -146,6 +146,31 @@ def initial_lowercase_then_upper_to_case_sensitive(
...
+def multiword(
+    resource: OntologyStringResource,
+) -> OntologyStringResource:
+    """If any synonym has more than one words, it's likely a noun phase and we should
+    mark all synonyms PROBABLE.
+
+    :param resource:
+    :return:
+    """
+    if any(" " in syn.text for syn in resource.original_synonyms):
```
Comment (paluchasz) [inline kazu/ontology_preprocessing/autocuration.py:158]: Could this also trigger if a synonym has an unnecessary whitespace that wasn't stripped or is that not possible?

For example `diabetes ` would trigger it but that is not the idea of this function, maybe we should use `.split` and count occurrences that way?

File: kazu/ontology_preprocessing/autocuration.py [hunk 2]
```diff
@@ -146,6 +146,31 @@ def initial_lowercase_then_upper_to_case_sensitive(
...
+            original_synonyms=frozenset(
+                dataclasses.replace(syn, mention_confidence=MentionConfidence.PROBABLE)
+                for syn in resource.original_synonyms
+            ),
+            alternative_synonyms=frozenset(
+                dataclasses.replace(syn, mention_confidence=MentionConfidence.PROBABLE)
+                for syn in resource.alternative_synonyms
+            ),
+        )
+    else:
```
Comment (paluchasz) [inline kazu/ontology_preprocessing/autocuration.py:170]: The `else` is unnecessary. Similarly in other functions in this file, maybe we could change there too.

See also https://docs.astral.sh/ruff/rules/superfluous-else-return/

**Review by paluchasz — CHANGES_REQUESTED** [review]
One other nit on this `autocurations.py` that maybe we could change: there are some missing return types on the `__init__` functions. It is common to have a `-> None` return type.

File: kazu/ontology_preprocessing/autocuration.py (hunk 1 above)
Comment (RichJackson) [reply kazu/ontology_preprocessing/autocuration.py:158]: it could... I'll add a strip to eliminate the possibility

**Review by paluchasz — APPROVED** [review]
